tensortrade/environments/trading_environment.py

```diff
@@ -76,7 +88,8 @@ def __init__(self,
         self.clock = Clock()
         self.action_space = None
         self.observation_space = None
-        self.viewer = None
+        self.render_mode = render_mode
+        self.viewer = PlotlyTradingChart(height=chart_height) if self.render_mode == 'human' else None
```

Comment:
We can allow it to default to `plotly` however.

Reply:
Yes (I left a suggestion about it in the class comment). My suggested approach:
- Create a standard renderer interface based on an abstract class and base all all renderers on it.
- Each renderer can have a different configuration but the same `render()` signature.
- Provide standard TensorTrade renderers (Matplotlib, Plotly, Logger, TensorBoard) and developers can create their own renderers conforming to the abstract class.
- The developer can "attach" any number of loggers to the environment and all can be used.
- If no renderer is provided, the environemnt will use a default renderer.

Something like this:
```
from tensortrade.environment.render import PlotlyTradingChart
from tensortrade.environment.render import LogRenderer
from tensortrade.environment.render import TensorBoardRenderer
from my_custom_renderer import CustomRenderer

# create and configure each renderer separately
renderer1 = PlotlyTradingChart(...)
renderer2 = LogRenderer(...)
renderer3 = TensorBoardRenderer(...)
renderer4 = CustomRenderer(...)

env = TradingEnvironment(
    feed=feed,
    portfolio=portfolio,
    action_scheme='managed-risk',
    reward_scheme='risk-adjusted',
...
    renderer=[renderer1, renderer2, renderer3, renderer4]
    # All renderers have the same render() signature.
    # The environment will loop over the list and invoke render() on them all.
)
```
(I'd use `renderer` instead of `view` because not every render will "view" the results.)

This will take some changes and a while to implement but the idea is worth it in my opinion. What do you think?

Reply:
On a second thought, TensorBoard wouldn't fit here so please ignore it in the previous example.

Reply:
I agree, we should make a standard, abstract `EnvironmentRenderer` with an abstract `render` function that can then be implemented by `PlotlyTradingChart`, `MatplotlibTradingChart`, `LogRenderer`, `CustomRenderer`, etc.

Then allowing for multiple `renderer`s is as simple as you have said.

Is this a change you could add to this PR?

Reply:
Please check the latest commit 2e30d00, it includes the following:
* An abstract renderer class and 3 renderers based on it:
  * PlotlyTradingChart: The same renderer modified for unified interface.
  * ScreenLogger: Replaces the hard-coded text inside the agent and provides control over progress messages.
  * FileLogger: Writes output to a file.
* Progress message removed from DQNAgent to use ScreenLogger optionally if required.
* An example file demonstrating the use of three renderers.
* A fix for the DQNAgent stopping after one episode (exists in A2CAgent too). But the default n_episode needs to be limited or it will run forever now. It could exit with another condition like reaching a certain `mean_reward`.
* It seems that epsilon was calculated incorrectly calculated in DQNAgent (and A2CAgent too). It should be fixed in DQNAgent now, please review.

Notice that only DQNAgent has been modified to use the above. After finailzing them we can apply to the others.

Reply:
I made some enhancements in 11c2406 with more detailed trade information on the chart.
![image](https://user-images.githubusercontent.com/9349772/74565692-78d43900-4f7a-11ea-92c2-aa3af8f8e398.png)